ggml/src/CMakeLists.txt

```
@@ -226,6 +226,9 @@ target_link_libraries(ggml PUBLIC ggml-base)
 
 if (CMAKE_SYSTEM_NAME MATCHES "Linux")
     target_link_libraries(ggml PRIVATE dl)
+    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
```

This is needless for recent GCC:

```suggestion
    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9")
```

FYI: We need `c++fs` (not `stdc++fs`) for Clang < 8 (`(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "8")`).